### PR TITLE
DOCS: SELECT/ SELECT INTO - Add RECURSIVE keyword to WITH clause 

### PR DESCRIPTION
**Pull request by mkiyama**
Updates for recursive WITH (recursive CTE queries) 
--SELECT command
--SELECT INTO command
--new Admin Guide topic "WITH Queries (Common Table Expressions)"
--Add new topic to AdminGuide TOC

Information based on PostgreSQL recursive WITH documentation.

NOTE: Documentation will not be merged until WITH RECURSIVE is enabled in GPDB.

DRAFT HTML
http://docs-gpdb-review-staging.cfapps.io/500Beta/ref_guide/sql_commands/SELECT.html
http://docs-gpdb-review-staging.cfapps.io/500Beta/admin_guide/query/topics/CTE-query.html

### CODE REVIEW

File: gpdb-doc/dita/admin_guide/query/topics/CTE-query.xml [hunk 1]
```diff
@@ -30,6 +30,11 @@ GROUP BY region, product;
     <p>The query could have been written without the <codeph>WITH</codeph> clause, but would have
       required two levels of nested sub-SELECTs. It is easier to follow with the
         <codeph>WITH</codeph> clause.</p>
+    <p>The optional <codeph>RECURSIVE</codeph> keyword can be enabled by setting the server
+      configuration parameter <codeph>gp_recursive_cte_prototype</codeph> to
+        <codeph>on</codeph>.<note type="warning">The <codeph>RECURSIVE</codeph> feature currently is
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/query/topics/CTE-query.xml]: Maybe change the warning to just "The RECURSIVE **keyword** is an experimental feature and is not intended for use in a production environment. Experimental features are subject to change without notice in future releases."

I definitely wouldn't add any guarantee like "The feature will promoted from experimental in a future release."

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 2]
```diff
@@ -4637,6 +4704,43 @@
...
+      <p>Controls the availability of the experimental keyword <codeph>RECURSIVE</codeph> in the
+          <codeph>WITH</codeph> clause of a <codeph>SELECT[ INTO]</codeph> command. The keyword
+        allows a subquery in the <codeph>WITH</codeph> clause of a <codeph>SELECT[ INTO]</codeph>
+        command to reference itself. The default value is <codeph>false</codeph>, the keyword is not
+        allowed in the <codeph>WITH</codeph> clause a <codeph>SELECT[ INTO]</codeph> command.</p>
+      <p>The parameter can be set for a database system, an individual database, or a session or
+        query.</p>
+      <p>For information about the experimental keyword <codeph>RECURSIVE</codeph>, see the
+          <codeph>SELECT</codeph> command.</p>
+      <note>This parameter will be removed when the <codeph>RECURSIVE</codeph> keyword is promoted
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: Repeat previous warning text and edit this to "This parameter will be removed **if** the RECURSIVE keyword feature is promoted from experimental in a future release."

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml [hunk 3]
```diff
@@ -114,9 +116,16 @@ with_query_name</varname> [ [AS] <varname>alias</varname> [( <varname>column_ali
...
           specified; if the list of column names is omitted, the names are inferred from the
           subquery. The primary query and the <codeph>WITH</codeph> queries are all (notionally)
-          executed at the same time. </p><p>If <codeph>RECURSIVE</codeph> is specified, it allows a
-          subquery to reference itself by name. Such a subquery, the <varname>select</varname>
-          portion of the <varname>with_query</varname> , must have the
+          executed at the same time. </p><p>The optional <codeph>RECURSIVE</codeph> keyword can be
+          enabled by setting the server configuration parameter
+            <codeph>gp_recursive_cte_prototype</codeph> to <codeph>true</codeph>. For information
+          about the parameter, see <xref href="../config_params/guc_config.xml#topic1"/>.<note
+            type="warning">The <codeph>RECURSIVE</codeph> feature currently is an experimental
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: Same edit here.

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml [hunk 4]
```diff
@@ -54,6 +54,8 @@ with_query_name</varname> [ [AS] <varname>alias</varname> [( <varname>column_ali
               ( <varname>column_definition</varname> [, ...] )
 <varname>from_item</varname> [NATURAL] <varname>join_type</varname> <varname>from_item</varname>
           [ON <varname>join_condition</varname> | USING ( <varname>join_column</varname> [, ...] )]</codeblock>
+      <p><b>NOTE:</b>
+        <sup>1</sup>The <codeph>RECURSIVE</codeph> keyword is an experimental feature.</p>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: Add "and is not recommended for production use."

**Comment by mkiyama** [comment]
@ivannovick @vraghavan78 
Please let us know if this not what you intended for experimental (development-level) features
or if the wording should change. (see the previous message)

Also, Venky, has somebody reviewed the recursive CTE content?

thanks

**Pull request merged**